sdk/resourcemanager/Azure.ResourceManager/src/ArmOperation.cs

```diff
@@ -3,13 +3,78 @@
 
 // <auto-generated/>
 
-#nullable disable
+#nullable enable
 
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
 
 namespace Azure.ResourceManager
 {
     /// <inheritdoc/>
-    public abstract class ArmOperation : Operation
-    {
+    public class ArmOperation : Operation
+{
+        private readonly OperationInternal? _operation;
+        private readonly RehydrationToken? _rehydrationToken;
+
+        /// <summary> Initializes a new instance of ArmOperation for mocking. </summary>
+        protected ArmOperation()
+        {
+        }
+
+        internal ArmOperation(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, Request request, Response response, OperationFinalStateVia finalStateVia, bool skipApiVersionOverride = false, string? apiVersionOverrideValue = null)
+        {
+            var nextLinkOperation = NextLinkOperationImplementation.Create(pipeline, request.Method, request.Uri.ToUri(), response, finalStateVia, skipApiVersionOverride, apiVersionOverrideValue);
+            _operation = new OperationInternal(nextLinkOperation, clientDiagnostics, response);
+        }
+
+        /// <summary> Create an instance of the <see cref="ArmOperation"/> class from rehydration. </summary>
+        /// <param name="client"> The client parameters to use in these operations. </param>
+        /// <param name="rehydrationToken"> The id of the ArmOperation. </param>
+        public ArmOperation(ArmClient client, RehydrationToken? rehydrationToken)
+        {
+            Argument.AssertNotNull(rehydrationToken, nameof(rehydrationToken));
+            Argument.AssertNotNull(client, nameof(client));
+
+            var nextLinkOperation = NextLinkOperationImplementation.Create(client.Pipeline, rehydrationToken);
+            // TODO: Do we need more specific OptionsNamespace, ProviderNamespace and OperationTypeName and possibly from id?
+            var clientDiagnostics = new ClientDiagnostics("Azure.ResourceManager", "Microsoft.Resources", client.Diagnostics);
+            _operation = new OperationInternal(nextLinkOperation, clientDiagnostics, null, rehydrationToken: rehydrationToken);
+            _rehydrationToken = rehydrationToken;
+        }
+
+        /// <inheritdoc />
+        public override RehydrationToken? GetRehydrationToken() => _rehydrationToken;
+
+#pragma warning disable CA1822
+        /// <inheritdoc />
+        public override string Id => throw new NotSupportedException();
```

Comment:
We should update this to return the Guid or null if this is a RELO pattern.

Reply:
Not quite sure how we want to handle this, let's say we will return `_operation.Id` here, then we need to prepare it in `OperationInternal`.
Given this Id is not implemented everywhere, shall we have a separate PR to fix them all?

Reply:
Yeah, I thought we cannot use the Id property for the rehydration token because it's already used for something else. Not, we want to throw from it?

Reply:
Added implementation for it.